File: backend/src/main/java/com/seniorcare/api/model/Medication.java
package com.seniorcare.api.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Table(name = "medications")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Medication {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @Column(name = "medication_name", nullable = false)
    private String medicationName;
    
    @Column(name = "dosage")
    private String dosage;
    
    @Column(name = "frequency")
    private String frequency;
    
    @Column(name = "start_date")
    private LocalDate startDate;
    
    @Column(name = "end_date")
    private LocalDate endDate;
    
    @Column(name = "instructions")
    private String instructions;
    
    @Column(name = "prescription_id")
    private String prescriptionId;
    
    @Column(name = "pharmacy_name")
    private String pharmacyName;
    
    @Column(name = "prescriber_name")
    private String prescriberName;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: watch-app/src/main/java/com/seniorcare/watch/domain/model/Medication.kt
package com.seniorcare.watch.domain.model

import java.util.Date

/**
 * 약물 정보 모델
 */
data class Medication(
    val id: Long,
    val medicationName: String,
    val dosage: String,
    val frequency: String,
    val startDate: Date,
    val endDate: Date? = null,
    val instructions: String? = null,
    val prescriptionId: String? = null,
    val pharmacyName: String? = null,
    val prescriberName: String? = null,
    val createdAt: Date = Date()
)

/**
 * 약물 복용 알림 모델
 */
data class MedicationReminder(
    val id: Long,
    val medicationId: Long,
    val medicationName: String,
    val dosage: String,
    val reminderTime: Date,
    val reminderStatus: ReminderStatus,
    val takenAt: Date? = null,
    val createdAt: Date = Date()
) {
    enum class ReminderStatus {
        SCHEDULED,  // 예정됨
        SENT,       // 알림 전송됨
        TAKEN,      // 복용함
        MISSED,     // 복용 안함
        CANCELED    // 취소됨
    }
}
